File: src/main/java/loxal/epvin/datastore/model/Vacation.java
/*
 * Copyright 2012 Alexander Orlov <alexander.orlov@loxal.net>. All rights reserved.
 */

package loxal.epvin.datastore.model;

import java.util.Date;

/**
 * @author Alexander Orlov <alexander.orlov@loxal.net>
 */
public class Vacation extends DatastoreEntity {
  private AppUser vacationer;

  public AppUser getVacationer() {
    return vacationer;
  }

  public void setVacationer(AppUser vacationer) {
    this.vacationer = vacationer;
  }

  private Date day;

  public Date getDay() {
    return day;
  }

  public void setDay(Date day) {
    this.day = day;
  }

  private boolean halfDay;

  public boolean isHalfDay() {
    return halfDay;
  }

  public void setHalfDay(boolean halfDay) {
    this.halfDay = halfDay;
  }
}

File: src/main/kotlin/loxal/epvin/datastore/model/Vacation.kt
/*
 * Copyright 2012 Alexander Orlov <alexander.orlov@loxal.net>. All rights reserved.
 */

package loxal.epvin.datastore.model

import java.util.Date

/**
 * @author Alexander Orlov <alexander.orlov@loxal.net>
 */
public class Vacation : DatastoreEntity() {
    public var vacationer: AppUser? = null

    public var day: Date? = null

    private var halfDay: Boolean = false

    public fun isHalfDay(): Boolean {
        return halfDay
    }

    public fun setHalfDay(halfDay: Boolean) {
        this.halfDay = halfDay
    }
}
